File: OP2_labs/lab4_py/Teacher.py
from Person import *

@dataclass
class TeacherSubject:
    name : str
    popularity : int

class Teacher(Person):
    def __init__(self, person):
        person[0] = person[0][2 : ]
        super().__init__(person[0])
        del person[0]
        self.subjects = []
        for row in person:
            self.subjects.append(TeacherSubject(row[ : row.rfind(' ')], int(row[row.rfind(' ') + 1 : ])))

    def getAvgPopularity(self):
        all : float = 0
        avg : float
        for subject in self.subjects:
            all += subject.popularity
        avg = all / len(self.subjects)
        return avg

File: OP2_labs/lab4_cpp/Teacher.cpp
#pragma once
#include "Teacher.h"

Teacher::Teacher(vector<string> text)
{
	text[0] = text[0].substr(2, text[0].length() - 2);
	birthday.year = stoi(text[0].substr(text[0].rfind('.') + 1, 4));
	text[0] = text[0].substr(0, text[0].rfind('.'));
	birthday.month = stoi(text[0].substr(text[0].rfind('.') + 1, 2));
	text[0] = text[0].substr(0, text[0].rfind('.'));
	birthday.day = stoi(text[0].substr(text[0].rfind(' ') + 1, 2));
	name = text[0].substr(0, text[0].rfind(' '));
	teacherSubject buffer;
	for (size_t i = 1; i < text.size(); i++)
	{
		buffer.name = text[i].substr(0, text[i].rfind(' '));
		buffer.popularity = stoi(text[i].substr(text[i].rfind(' ') + 1, text[i].length() - text[i].rfind(' ')));
		subjects.push_back(buffer);
	}
}
float Teacher::getAvgPopularity()
{
	float all = 0, avg;
	for (teacherSubject subject : subjects) { all += subject.popularity; }
	avg = all / subjects.size();
	return avg;
}